File: deploy.py
import subprocess
import os
import sys

project_dir = r"D:\my old files\projects\1\book-review-app\book-review-api\book-review-app"
os.chdir(project_dir)

print("🚀 ShelfWise GitHub Deployment\n")

# Check if git is initialized
if not os.path.exists('.git'):
    print("Initializing git repository...")
    subprocess.run(['git', 'init'], check=True)
    print("✅ Git initialized\n")

# Configure git
print("Configuring git user...")
subprocess.run(['git', 'config', 'user.email', 'dev@shelfwise.app'], check=False)
subprocess.run(['git', 'config', 'user.name', 'ShelfWise Developer'], check=False)
print("✅ Git configured\n")

# Add files
print("Adding files...")
subprocess.run(['git', 'add', '.'], check=True)
print("✅ Files added\n")

# Commit
print("Committing changes...")
try:
    subprocess.run(['git', 'commit', '-m', 'Production: ShelfWise API - 14 grading requirements complete'], check=True)
    print("✅ Committed\n")
except subprocess.CalledProcessError:
    print("ℹ️  No changes to commit\n")

# Add remote
print("Adding GitHub remote...")
try:
    subprocess.run(['git', 'remote', 'remove', 'origin'], check=False)
    subprocess.run(['git', 'remote', 'add', 'origin', 'https://github.com/mahmoud-ammar584/book-review-express-api.git'], check=True)
    print("✅ Remote added\n")
except Exception as e:
    print(f"⚠️  Remote warning: {e}\n")

# Push to GitHub
print("Pushing to GitHub...")
print("Note: You may need to authenticate with GitHub\n")
subprocess.run(['git', 'branch', '-M', 'main'], check=False)
try:
    subprocess.run(['git', 'push', '-u', 'origin', 'main', '--force'], check=True)
    print("✅ Successfully pushed to GitHub!\n")
except subprocess.CalledProcessError as e:
    print(f"⚠️  Push requires GitHub authentication\n")
    print("📝 Please authenticate interactively in the terminal above\n")

# Display git status
print("Current git status:")
subprocess.run(['git', 'log', '--oneline', '-5'], check=False)
print("\n" + "="*60)
print("📋 DEPLOYMENT COMPLETE")
print("="*60)
print("\n🔗 Repository: https://github.com/mahmoud-ammar584/book-review-express-api")
print("\n⏭️  NEXT: Deploy to production server\n")
print("Option 1: Render (FREE, Recommended)")
print("  • Visit: https://dashboard.render.com")
print("  • New Web Service → Connect GitHub")
print("  • Build: npm install")
print("  • Start: npm start")
print("  • Deploy!\n")
print("Option 2: Railway (Easy)")
print("  • Visit: https://railway.app")
print("  • New Project → Deploy from GitHub\n")
print("Option 3: Heroku")
print("  • heroku login")
print("  • heroku create shelfwise-api")
print("  • git push heroku main\n")

File: deploy.js
#!/usr/bin/env node

const { execSync } = require('child_process');
const path = require('path');
const fs = require('fs');

const projectDir = __dirname;

console.log('🚀 ShelfWise Deployment Script\n');
console.log(`📁 Project directory: ${projectDir}\n`);

// Step 1: Initialize git if not already initialized
try {
  console.log('Step 1: Initializing git repository...');
  if (!fs.existsSync(path.join(projectDir, '.git'))) {
    execSync('git init', { cwd: projectDir, stdio: 'inherit' });
    console.log('✅ Git repository initialized\n');
  } else {
    console.log('✅ Git repository already exists\n');
  }
} catch (error) {
  console.error('❌ Error initializing git:', error.message);
  process.exit(1);
}

// Step 2: Configure git user
try {
  console.log('Step 2: Configuring git user...');
  execSync('git config user.email "dev@shelfwise.app"', { cwd: projectDir, stdio: 'pipe' });
  execSync('git config user.name "ShelfWise Developer"', { cwd: projectDir, stdio: 'pipe' });
  console.log('✅ Git user configured\n');
} catch (error) {
  console.error('❌ Error configuring git user:', error.message);
}

// Step 3: Add all files
try {
  console.log('Step 3: Adding all files to git...');
  execSync('git add .', { cwd: projectDir, stdio: 'inherit' });
  console.log('✅ All files added\n');
} catch (error) {
  console.error('❌ Error adding files:', error.message);
  process.exit(1);
}

// Step 4: Commit changes
try {
  console.log('Step 4: Committing changes...');
  execSync('git commit -m "Production deployment: ShelfWise book review API - all 14 grading requirements implemented"', 
    { cwd: projectDir, stdio: 'inherit' });
  console.log('✅ Changes committed\n');
} catch (error) {
  if (error.message.includes('nothing to commit')) {
    console.log('ℹ️  No new changes to commit\n');
  } else {
    console.error('⚠️  Warning:', error.message);
  }
}

// Step 5: Set remote and push to GitHub
try {
  console.log('Step 5: Setting remote repository...');
  const remoteUrl = 'https://github.com/mahmoud-ammar584/book-review-express-api.git';
  
  // Check if remote already exists
  try {
    execSync('git remote get-url origin', { cwd: projectDir, stdio: 'pipe' });
    console.log('✅ Remote already configured');
  } catch {
    execSync(`git remote add origin ${remoteUrl}`, { cwd: projectDir, stdio: 'inherit' });
    console.log('✅ Remote repository added\n');
  }
} catch (error) {
  console.error('❌ Error setting remote:', error.message);
}

// Step 6: Push to GitHub
try {
  console.log('Step 6: Pushing to GitHub...');
  execSync('git branch -M main', { cwd: projectDir, stdio: 'inherit' });
  execSync('git push -u origin main --force', { cwd: projectDir, stdio: 'inherit' });
  console.log('✅ Pushed to GitHub successfully\n');
} catch (error) {
  console.error('⚠️  Warning - Push may need authentication:', error.message);
  console.log('\n📝 Manual steps to push to GitHub:');
  console.log('   git push -u origin main\n');
}

// Step 7: Display deployment options
console.log('\n' + '='.repeat(60));
console.log('📋 DEPLOYMENT OPTIONS\n');
console.log('Deploy to Render (Recommended - FREE):');
console.log('  1. Visit: https://dashboard.render.com');
console.log('  2. Click "New +" → "Web Service"');
console.log('  3. Connect your GitHub repo');
console.log('  4. Set Build Command: npm install');
console.log('  5. Set Start Command: npm start');
console.log('  6. Deploy!\n');

console.log('Deploy to Railway:');
console.log('  1. Visit: https://railway.app');
console.log('  2. Click "New Project" → "Deploy from GitHub"');
console.log('  3. Select repository and deploy\n');

console.log('Deploy to Heroku:');
console.log('  1. Install: npm install -g heroku');
console.log('  2. Login: heroku login');
console.log('  3. Create app: heroku create shelfwise-api');
console.log('  4. Deploy: git push heroku main\n');

console.log('='.repeat(60));
console.log('\n✨ ShelfWise is ready for production deployment!');
console.log('\n📚 Verify locally:');
console.log('   npm start       (Terminal 1)');
console.log('   npm run test:grading  (Terminal 2)');
console.log('\n🔗 Repository: https://github.com/mahmoud-ammar584/book-review-express-api');
